CRM/SMS/Form/Schedule.php

```diff
@@ -111,24 +120,23 @@ public function buildQuickform() {
    *   set.
    */
   public static function formRule($params, $files, $self) {
-    if (!empty($params['_qf_Schedule_submit'])) {
 
+    if (!empty($params['_qf_Schedule_submit'])) {
       CRM_Core_Session::setStatus(ts("Your Mass SMS has been saved. Click the 'Continue' action to resume working on it."), ts('Saved'), 'success');
       $url = CRM_Utils_System::url('civicrm/mailing/browse/unscheduled', 'scheduled=false&reset=1&sms=1');
       CRM_Utils_System::redirect($url);
     }
-    if (isset($params['now']) || CRM_Utils_Array::value('_qf_Schedule_back', $params) == ts('Previous')) {
+
+    if ((isset($params['send_option']) && $params['send_option'] == 'send_immediate') || CRM_Utils_Array::value('_qf_Schedule_back', $params) == ts('Previous')) {
       return TRUE;
     }
 
-    if (CRM_Utils_Date::format(CRM_Utils_Date::processDate($params['start_date'],
-        $params['start_date_time']
-      )) < CRM_Utils_Date::format(date('YmdHi00'))
-    ) {
+    if (CRM_Utils_Date::format(CRM_Utils_Date::processDate($params['start_date'])) < CRM_Utils_Date::format(date('YmdHi00'))) {
```

Comment:
I don't think we need processDate here?

Reply:
Agreed, I need to look at that whole line again.